sw/device/lib/base/log.c

```diff
@@ -64,11 +69,22 @@ void base_log_internal_core(log_severity_t severity, const char *file_name,
  * Logs `format` and the values that following in an efficient, DV-testbench
  * specific way.
  *
- * @param severity the log severity.
- * @param format a format string, as described in print.h. This must be a string
- * literal.
+ * @param log the log_fields_t struct that holds the log fields.
+ * @param nargs the number of arguments passed to the format string.
  * @param ... format parameters matching the format string.
  */
-void base_log_internal_dv(log_severity_t severity, const char *format, ...) {
-  // Do nothing, for now.
+void base_log_internal_dv(const log_fields_t *log, int nargs, ...) {
+  va_list args;
+  uint32_t value;
+
+  va_start(args, nargs);
+  // The first arg in the va_list is the format string - discard it.
+  value = va_arg(args, uint32_t);
+
+  REG32(SW_LOG_DV_ADDR) = (uintptr_t)log;
+  for (int i = 0; i < nargs; ++i) {
+    value = va_arg(args, uint32_t);
+    REG32(SW_LOG_DV_ADDR) = value;
+  }
```

Comment:
`REG32` is going away. Just create an `mmio_region`: it's a free, inlined operation.

Reply:
updated.